src/api/utils/index.js

```diff
@@ -1,31 +1,117 @@
+import 'babel-polyfill';
 import imm from 'object-path-immutable';
+import Registry from '../../registry';
 
 export apiClient from './api-client';
 
+/**
+ * @module
+ * @private
+ */
+
+/**
+ * @example
+ * incrementProperty(state, 'isReading')
+ * @param {Object} state Object on which the key is incremented
+ * @param {string} key Key to increment
+ */
 export const incrementProperty = (state, key) => (
   imm.set(state, key, state[key] + 1)
 );
 
+/**
+ * @example
+ * decrementProperty(state, 'isReading')
+ * @param {Object} state Object on which the key is decremented
+ * @param {string} key Key to decrement
+ */
 export const decrementProperty = (state, key) => (
   imm.set(state, key, state[key] - 1)
 );
 
-export const resourcesArray = (data) => (
+/**
+ * @param {Array|Object} data
+ */
+export const resourcesArray = (data = []) => (
   Array.isArray(data) ? data : [data]
 );
 
+/**
+ * @example
+ * const obj = { foo: bar: { baz: 'hi' } }
+ * hasProperty(obj, 'foo', 'bar', 'baz') // => 'hi'
+ * hasProperty(obj, 'a', 'b', 'c') // => undefined
+ *
+ * @param {Object} object Object to traverse
+ * @param {String[]} keys Keys to look up
+ */
+export const hasProperty = (object, ...keys) => (
```

Comment:
I didn't want to introduce a dependency for such a small thing, since it can conflict with people's own dependencies.

Currently, our dependencies are:
1. `axios`
2. `object-path-immutable`
3. `babel-polyfill`
4. `yup`

I'd like to reduce this to just the first two if possible, with peer dependencies of `redux` and `redux-thunk`.